packages/controllers/src/controllers/EventsController.ts

```diff
@@ -116,7 +99,41 @@ export const EventsController = {
       'SOCIAL_LOGIN_SUCCESS'
     ]
     if (OptionsController.state.features?.analytics || MANDATORY_EVENTS.includes(data.event)) {
-      EventsController._sendAnalyticsEvent(state)
+      EventsController._setPendingEvent(state)
+    }
+    // Calling this function here to make sure document is ready and defined before subscribing to visibility change
+    this._subscribeToVisibilityChange()
+  },
+
+  _submitPendingEvents() {
+    if (state.pendingEvents.length === 0) {
+      return
     }
+    try {
+      api.sendBeacon({
+        path: '/batch',
+        params: EventsController.getSdkProperties(),
+        body: state.pendingEvents
+      })
+      state.reportedErrors['FORBIDDEN'] = false
```

Comment:
What do we do with FORBIDDEN error when it's set to true?

Reply:
nothing really, even in the current version - it was just stays there